transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -47,6 +49,7 @@
             new IllegalStateException("Channel is unhealthy not offering it back to pool"),
             SimpleChannelPool.class, "releaseAndOffer(...)");
 
+    protected final ConcurrentMap<Channel, Channel> allChannels = PlatformDependent.newConcurrentHashMap();
```

Comment:
Hi @normanmaurer, thanks for your comments.

ChannelGroup is too heavy for this scenario.

What we want is
1. when acquiring, new channels shall be added to the container
2. when releasing, inactive channels shall be removed from container if necessary
3. we don't need deregister, disconnect, write, flush, and such things of ChannelGroup

Yours sincerely,
Xiaoshuang LU